detection/detect_objects.cpp

```diff
@@ -36,22 +37,29 @@ void filterLargest(vector<Circle>* key_circ, vector<Circle>* circ, int max_count
 	}
 }
 
-void detectObjects(Mat* frame, vector<Point2f>* centers, int obj_count = 5) {
+int shiftHue(int hue) {
+	return (hue + COLOR_OFFSET) % 180;
+}
+
+void detectObjects(Mat* frame, vector<Point2f>* centers, int max_obj_count = 5) {
 	cuda::GpuMat d_frame;
 	Mat h_frame, h_red_blobs, h_lwr_red_blobs, h_upr_red_blobs, h_grn_blobs;
-	Scalar lowest_red(0, 150, 150), lower_red(20, 255, 255);
-	Scalar upper_red(160, 150, 150), uppest_red(180, 255, 255);
-	Scalar lower_grn(50, 80, 80), upper_grn(90, 255, 255);
+	Scalar lower_red(shiftHue(160), 150, 150), upper_red(shiftHue(20), 255, 255);
+	Scalar lower_grn(shiftHue(50), 80, 80), upper_grn(shiftHue(90), 255, 255);
 	Scalar red(0, 0, 255), grn(0, 255, 0);
-	
+
 	d_frame.upload(*frame);
 	cuda::cvtColor(d_frame, d_frame, COLOR_BGR2HSV);
 	d_frame.download(h_frame);
 
-	inRange(h_frame, lowest_red, lower_red, h_lwr_red_blobs);
-	inRange(h_frame, upper_red, uppest_red, h_upr_red_blobs);
+	for (int i = 0; i < h_frame.rows; i++) {
```

Comment:
Forgot to mention this in a TODO.... I'd like to look in to a better of traversing the whole matrix.

Reply:
Is this the actual process of shifting the colors?

Reply:
Indeed it is.

Reply:
```Make the conversion: cvCvtColor(imageBgr, imageHsv, CV_RGB2HSV);

Note, RGB vs Bgr are purposefully being crossed.

This way, red color will be treated in a blue channel and will be centered around 170. There would also be a flip in direction, but that is OK as long as you know to expect it.```
https://stackoverflow.com/a/31084879

This could be an option.

Reply:
This also would allow us to the the gpu to speed the shifting up, since `cvtColor` is also a gpu function